packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,617 @@
+        ></nimble-mapping-mention-user>
+    </nimble-rich-text-mention-issues>
+</nimble-rich-text-editor>
+```
+
+#### Alternatives
+
+Slot based approach: All user information will be communicated through `nimble-list-option` elements. These elements will
+contain the user's ID in their `value` attribute and the user's name in their text content. When the **"@"** symbol is
+inserted into the editor, these elements will be presented as a dropdown list.
+
+Pros:
+
+1. When the client does not specify a `value` attribute, the options are sent without a user's ID, and in such cases,
+   the value attribute is derived from the content of the option.
+
+Cons:
+
+1. It is impossible to have multiple type of mentions when `nimble-list-option` is used in the default slot. If we use
+   slot names to differentiate between various type of mentions, the rich text editor needs to be aware of all the type
+   of mentions like user slot name, issue slot name etc, through an attribute value.
+2. The slot-based approach is not configurable, meaning that if a user wishes to customize the entire user mention with
+   some arbitrary value, the only viable option is to include it within the options. In contrast, the configuration-based
+   approach readily allows for the inclusion of arbitrary values in the parent container element, making customization more
+   straightforward. For an example, in future if we are configuring the `character` to trigger the user mention from the client,
+   it is simply to add it like below,
+    ```html
+    <nimble-rich-text-mention-users character="@">
+        <nimble-mapping-mention-users
+            key="user-id-1"
+            text="username-1"
+        ></nimble-mapping-mention-users>
+        <nimble-mapping-mention-users
+            key="user-id-2"
+            text="username-2"
+        ></nimble-mapping-mention-users>
+        <nimble-mapping-mention-users
+            key="user-id-3"
+            text="username-3"
+        ></nimble-mapping-mention-users>
+    </nimble-rich-text-mention-users>
+    ```
+    If it is a slot based approach, it is required to provide the information to every options as we don't know that it is a user mention
+    or other mention
+    ```html
+    <nimble-mapping-mention-users
+        key="user-id-1"
+        text="username-1"
+        character="@"
+    ></nimble-mapping-mention-users>
+    <nimble-mapping-mention-users
+        key="user-id-2"
+        text="username-2"
+        character="@"
+    ></nimble-mapping-mention-users>
+    <nimble-mapping-mention-users
+        key="user-id-3"
+        text="username-3"
+        character="@"
+    ></nimble-mapping-mention-users>
+    ```
+3. The UI of the list option is completely dependent on the client application.
+4. All the keyboard and mouse interactions within the dropdown list should be handled internally within the editor as `nimble-list-option`
+   does not support any.
+
+#### Client Usage Guidance on Filtered Users:
```

Comment:
When each keystroke could trigger an expensive operation like a network request, it's good practice to "debounce" the events to limit the number of requests (e.g. allow at most one request per second). We should give guidance about who is responsible for doing that.

Since the SLE use case will have access to all the users in the client and won't need to do debouncing, I lean towards saying that the Nimble component doesn't need to implement that logic. Rather we should just add client usage guidance that reminds clients that they should do it if necessary.

Reply:
Makes sense. I have updated the client usage guidance as suggested.